SC_redesign/Crop_and_Soil/soil/phosphorus_cycling/soluble_phosphorus.py

```diff
@@ -0,0 +1,227 @@
+
+        top_layer_soil_phosphorus_concentration = self._determine_soil_phosphorus_concentration(
+            self.data.soil_layers[0].labile_phosphorus_content, self.data.soil_layers[0].bulk_density,
+            self.data.soil_layers[0].layer_thickness, field_size)
+        extraction_coefficient = 0.005
+        top_layer_dissolved_reactive_phosphorus_runoff = top_layer_soil_phosphorus_concentration * \
+            extraction_coefficient * runoff_in_liters_per_hectare * (10 ** (-6))
+
+        adjusted_phosphorus_runoff = min(self.data.soil_layers[0].labile_phosphorus_content,
+                                         top_layer_dissolved_reactive_phosphorus_runoff)
+        self.data.soil_layers[0].labile_phosphorus_content -= adjusted_phosphorus_runoff
+        self.data.annual_soil_phosphorus_runoff += adjusted_phosphorus_runoff * field_size
+
+    # --- Static methods ---
+    @staticmethod
+    def _determine_isotherm_slope(percent_clay_content: float) -> float:
+        """Calculates the slope of the linear phosphorus sorption isotherm.
+
+        Parameters
+        ----------
+        percent_clay_content : float
+            Percent clay content of a soil layer, expressed in the range [0, 100] (unitless)
+
+        Returns
+        -------
+        float
+            The slope of the phosphorus sorption isotherm (unitless)
+
+        References
+        ----------
+        APLE Theoretical Documentation eqn. [15]
+
+        """
+        return 173.51 * (percent_clay_content / 100) + 8.48
+
+    @staticmethod
+    def _determine_isotherm_intercept(isotherm_slope: float) -> float:
+        """Calculates the intercept of the linear phosphorus sorption isotherm.
+
+        Parameters
+        ----------
+        isotherm_slope : float
+            The slope of the phosphorus sorption isotherm (unitless)
+
+        Returns
+        -------
+        float
+            The intercept of the phosphorus sorption isotherm (unitless)
+
+        References
+        ----------
+        APLE Theoretical Documentation eqn. [16]
+
+        """
+        return 4.726 * isotherm_slope - 8.97
+
+    @staticmethod
+    def _determine_soil_phosphorus_concentration(labile_phosphorus: float, bulk_density: float,
+                                                 layer_thickness: float, field_size: float) -> float:
+        """Calculates the concentration of phosphorus in a soil layer.
+
+        Parameters
+        ----------
+        labile_phosphorus : float
+            Labile phosphorus content of this soil layer (kg phosphorus per ha)
+        bulk_density : float
+            Bulk density of the soil layer (Megagram per cubic meter)
+        layer_thickness : float
+            Thickness of the soil layer (mm)
+        field_size : float
+            Area of the field (ha)
+
+        Returns
+        -------
+        float
+            The concentration of phosphorus in the soil layer (mg phosphorous per kg soil)
```

Comment:
```suggestion
            The concentration of phosphorus in the soil layer (mg phosphorous/kg soil)
```
Maybe this would be more in line with how we document units in other files? e.g. (kg/ha)

Reply:
Done